include/aikido/planner/ompl/detail/OMPLConfigurationToConfigurationPlanner-impl.hpp

```diff
@@ -0,0 +1,200 @@
+#include "aikido/planner/ompl/GeometricStateSpace.hpp"
+#include "aikido/planner/ompl/MotionValidator.hpp"
+#include "aikido/planner/ompl/Planner.hpp"
+#include "aikido/statespace/GeodesicInterpolator.hpp"
+
+namespace aikido {
+namespace planner {
+namespace ompl {
+
+using aikido::constraint::dart::createProjectableBounds;
+using aikido::constraint::dart::createSampleableBounds;
+using aikido::constraint::dart::createTestableBounds;
+using aikido::constraint::dart::createProjectableBounds;
+using aikido::distance::createDistanceMetric;
+using aikido::statespace::dart::MetaSkeletonStateSpace;
+
+//==============================================================================
+template <class PlannerType>
+OMPLConfigurationToConfigurationPlanner<PlannerType>::
+    OMPLConfigurationToConfigurationPlanner(
+        statespace::ConstStateSpacePtr stateSpace,
+        common::RNG* rng,
+        statespace::ConstInterpolatorPtr interpolator,
+        distance::DistanceMetricPtr dmetric,
+        constraint::SampleablePtr sampler,
+        constraint::TestablePtr boundsConstraint,
+        constraint::ProjectablePtr boundsProjector,
+        double maxDistanceBtwValidityChecks)
+  : ConfigurationToConfigurationPlanner(std::move(stateSpace), rng)
+  , mInterpolator(std::move(interpolator))
+  , mBoundsConstraint(std::move(boundsConstraint))
+  , mBoundsProjector(std::move(boundsProjector))
+  , mMaxDistanceBtwValidityChecks(maxDistanceBtwValidityChecks)
+{
+  if (!mInterpolator)
+    mInterpolator = std::make_shared<aikido::statespace::GeodesicInterpolator>(
+        mStateSpace);
+
+  if (!dmetric)
+    dmetric = std::move(createDistanceMetric(mStateSpace));
+
+  // TODO (avk): The constraint namespace functions only work with
+  // metaskeletonstatespace
+  const auto metaskeletonStatespace
+      = std::dynamic_pointer_cast<const MetaSkeletonStateSpace>(mStateSpace);
+
+  if (!metaskeletonStatespace
+      && (!sampler || !boundsConstraint || !boundsProjector))
+  {
+    throw std::invalid_argument(
+        "[OMPLPlanner] Either statespace must be metaSkeletonStateSpace or all "
+        "of sampler, boundsContraint and boundsProjector must be provided.");
+  }
+
+  if (!sampler)
+  {
+    if (!rng)
+      throw std::invalid_argument(
+          "[OMPLPlanner] Both of sampler and rng cannot be nullptr.");
+    sampler = std::move(
+        createSampleableBounds(metaskeletonStatespace, rng->clone()));
+  }
+
+  if (!mBoundsConstraint)
+    mBoundsConstraint = std::move(createTestableBounds(metaskeletonStatespace));
+
+  if (!mBoundsProjector)
+    mBoundsProjector
+        = std::move(createProjectableBounds(metaskeletonStatespace));
+
+  // Geometric State space
+  auto sspace = ompl_make_shared<GeometricStateSpace>(
+      mStateSpace,
+      std::make_shared<aikido::statespace::GeodesicInterpolator>(mStateSpace),
+      std::move(dmetric),
+      std::move(sampler),
+      mBoundsConstraint,
+      std::move(mBoundsProjector));
```

Comment:
We don't need to have `mBoundsProjector` as a member variable because it's been moved. It seems we could convert this as a local variable in this constructor implementation.